cpp/include/cudf/rolling.hpp

```diff
@@ -379,6 +401,125 @@ std::unique_ptr<column> grouped_time_range_rolling_window(
   std::unique_ptr<aggregation> const& aggr,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief  Applies a grouping-aware, value range-based rolling window function to the values in a
+ *         column.
+ *
+ * This function aggregates rows in a window around each element of a specified `input` column.
+ * The window is determined based on the values of an ordered `orderby` column, and on the values
+ * of a `preceding` and `following` scalar representing an inclusive range of orderby column values.
+ *
+ *   1. The elements of the `input` column are grouped into distinct groups (e.g. the result of a
+ *      groupby), determined by the corresponding values of the columns under `group_keys`. The
+ *      window-aggregation cannot cross the group boundaries.
+ *   2. Within a group, with all rows sorted by the `orderby` column, the aggregation window
+ *      for a row at index `i` is determined as follows:
+ *      a) If `orderby` is ASCENDING, aggregation window for row `i` includes all `input` rows at
+ *         index `j` such that:
+ *         @code{.pseudo}
+ *           (orderby[i] - preceding) <= orderby[j] <= orderby[i] + following
+ *         @endcode
+ *      b) If `orderby` is DESCENDING, aggregation window for row `i` includes all `input` rows at
+ *         index `j` such that:
+ *         @code{.pseudo}
+ *           (orderby[i] + preceding) >= orderby[j] >= orderby[i] - following
+ *         @endcode
+ *
+ * Note: This method requires that the rows are presorted by the group keys and orderby column
+ * values.
+ *
+ * The window intervals are specified as scalar values appropriate for the orderby column.
+ * Currently, only the following combinations of `orderby` column type and range types
+ * are supported:
+ *   1. If `orderby` column is a TIMESTAMP, the `preceding`/`following` windows are specified
+ *      in terms of lower resolution `DURATION` scalars.
```

Comment:
This is wrong: I thought this had been fixed already.

In the current implementation, we no longer support duration-types with lower resolution than the orderby column. The resolutions must be identical. `DURATION_DAYS` for `TIMESTAMP_DAYS`, `DURATION_SECONDS` for `TIMESTAMP_SECONDS`, `INT32` for `INT32`.

I will correct this shortly.

Reply:
Done.